source/generate_cids/generate_context_groups.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Encoding required to deal with 'micro' character
+"""Script for auto-generating DICOM SR context groups from FHIR JSON value set
+resources."""
+import ftplib
+import json
+import logging
+import os
+import re
+from io import BytesIO
+
+logger = logging.getLogger(__name__)
+
+# The list of scheme designators is not complete.
+# For full list see table 8-1 in part 3.16 chapter 8:
+# http://dicom.nema.org/medical/dicom/current/output/chtml/part16/chapter_8.html#table_8-1
+FHIR_SYSTEM_TO_DICOM_SCHEME_DESIGNATOR = {
+    'http://snomed.info/sct': 'SCT',
+    'http://snomed.info/srt': 'SRT',
+    'http://dicom.nema.org/resources/ontology/DCM': 'DCM',
+    'http://loinc.org': 'LN',
+    'http://www.radlex.org': 'RADLEX',
+    'http://sig.biostr.washington.edu/projects/fm/AboutFM.html': 'FMA',
+    'http://www.nlm.nih.gov/mesh/meshhome.html': 'MSH',
+    'http://ncit.nci.nih.gov': 'NCIt',
+    'http://unitsofmeasure.org': 'UCUM',
+    'http://hl7.org/fhir/sid/ndc': 'NDC',
+    'urn:iso:std:iso:11073:10101': 'MDC',
+    'doi:10.1016/S0735-1097(99)00126-6': 'BARI',
+    'http://www.nlm.nih.gov/research/umls': 'UMLS',
+    'http://pubchem.ncbi.nlm.nih.gov': 'PUBCHEM_CID',
+    'http://braininfo.rprc.washington.edu/aboutBrainInfo.aspx#NeuroNames': 'NEU',
+    'http://www.itis.gov': 'ITIS_TSN',
+}
+
+if __name__ == '__main__':
+
+    logging.basicConfig(level=logging.INFO)
+
+    ftp_host = 'medical.nema.org'
+    logger.info('log into FTP server "{}"'.format(ftp_host))
+    ftp = ftplib.FTP(ftp_host, timeout=60)
+    ftp.login('anonymous')
+
+    ftp_path = 'medical/dicom/resources/valuesets/fhir/json'
+    logger.info('list files in directory "{}"'.format(ftp_path))
+    ftp_files = ftp.nlst(ftp_path)
+    ftp_url = 'ftp://{host}/{path}'.format(host=ftp_host, path=ftp_path)
+
+    cid_pattern = re.compile('^dicom-cid-([0-9]+)-[a-zA-Z]+')
+    for ftp_filepath in ftp_files:
```

Comment:
FHIR JSON includes `version` - how about we also include it in the pydicom CIDs?